### ISSUE TITLE
Bidirectional relationship problem

### ISSUE THREAD

**Issue by george-fb** [pods-framework/pods#3361]
Hello!

I have a problem with bidirectional relationship.

I created a relationship field for posts:
![basic](http://i.imgur.com/NcSyyA4.png)
![additional](http://i.imgur.com/SPqH3od.png)

I create a post "Post 1" (no relationship), I create "Post 2" related to "Post 1".
All is fine till here "Post 1" is related to "Post 2" and vice versa.
Now if I add "Post 3" and link it to "Post 1" and "Post 2", "Post 3" will be related to both, but "Post 1" and "Post 2" will be related to themselves too now, and I can not remove the relation to themselves.

So:
Post 1 related to -> Post 1, Post 2 and Post 3
Post 2 related to -> Post 1, Post 2 and Post 3
Post 3 related to -> Post 1, Post 2

I tested this on a fresh WP 4.4.1 installation with only pods installed.

**Comment by jimtrue** [comment]
This is actually not a bug. When you create a 'bidirectional' relationship, that means you're connecting it on 'both sides' of the relationship. So if you relate post 1 to post 2, post 2 is automatically related to post 1 as well. If you relate post 3 to post 1 and post 2, post 3 will also automatically be related to post 1 and post 2, so now post 1 is related to the others.

If you don't want that kind of behavior, remove the bidirectional field setting and that way the relationship won't go in both directions and create the circular relationship you've created.

**Comment by george-fb** [comment]
The problem is when i relate post 3 to post 1 and 2 then post 1 becomes related to itself same for post 2.

**Comment by jimtrue** [comment]
@george-fb I actually just recreated your setup and I'm not getting unusual post relating to themselves issue you're describing above.

Try clearing the Cache, you might have been going back and forth and got some unusual transients stuff in your database. Pods Admin, Settings, Clear Cache.

Post 1, no relationships
Add Post 2, related to Post 1: Post 2 has Post 1 and Post 1 now has Post 2 (expected)
Add Post 3, related to Post 1 and Post 2: Post 1 now has Post 2 and Post 3 (expected) and Post 2 now has Post 1 and Post 3 (also expected).

**Comment by george-fb** [comment]
I reinstalled WP and pods and i have the same issue.
Here is a video https://www.youtube.com/watch?v=E0yP7ISXUe4

**Comment by jimtrue** [comment]
Weird, now I'm getting it. It didn't occur on my end until I added a Pods Template to the Post to show the related posts, though I can't imagine that would be in any way related. 

But yes, I'm now getting the exact same odd behavior you're getting (even though when I created the  And it didn't happen until Post 3 was added with the relationship to 1 and 2. I still think this has to be in some way related to creating a circular relationship, but we're going to have to look at this in the debugging to be for sure.

I've also deleted entries, deleted Pod and Cleared Cache and the exact same issue on the third post which wasn't happening when I first created and replied to you this morning. 

Pushing up the line.

**Comment by jimtrue** [comment]
@pglewis I think this is creating some kind of a circular reference. Without digging into wp_podsrel, I can't be sure how it's doing it, but think this is definitely an edge case for relating a pod back to itself. I've only tested this with Extended Posts. Haven't ran through and tested with other Post Types to verify it's reproducible in all of those situations, but I can do that. 

I think we'd be getting hit with a lot more of these if that was the case though.

**Comment by george-fb** [comment]
I tested this on a CPT and the same happens.

**Comment by jimtrue** [comment]
Didn't realize you were doing this on an ACT; I did mine on CPT and I saw the same issue. We've verified it's reproducible, now we just need to find out why.

**Comment by george-fb** [comment]
No, it was on the default post type. Guess you were talking about ACT above.

**Comment by jimtrue** [comment]
You extended Posts, so that's only a Post Type, not Advanced Content Types. But I am curious to determine if we have the same issue with other Post Types.

**Comment by quasel** [comment]
See #4375 additional stuff going on it seems 
Same situation: 
> It results in generating many records in table wp_podsrel
